ukis_pysat/data.py

```diff
@@ -157,8 +140,12 @@ def query_metadata(self, platform, date, aoi, cloud_cover=None):
             meta_src = self.api.query(area=self.prep_aoi(aoi).wkt, date=date, platformname=platform.value, **kwargs,)
             meta_src = self.api.to_geojson(meta_src)["features"]
 
-        # construct MetadataCollection from list of Metadata objects
-        return MetadataCollection([self.construct_metadata(meta_src=m, platform=platform) for m in meta_src])
+        if not self.catalog:
+            self.init_catalog()
+
+        for item in meta_src:
+            self.catalog.add_item(self.construct_metadata(meta_src=item, platform=platform))
+        return self.catalog
 
     def query_metadata_srcid(self, platform, srcid):

```

Comment:
it would make more sense to return an item here

Reply:
Personally, I woud vote for returning a catalog here. IMHO this is a reasonable tradeoff as we then always handle catalogs.